File: comicboxd/app/model/series.go
package model

import "github.com/google/uuid"

// Series is a series in the database
type Series struct {
	Name     string    `json:"name"   db:"name"`
	Total    int32     `json:"total"  db:"total"`
	Read     int32     `json:"read"   db:"read"`
	UserID   uuid.UUID `json:"-"      db:"user_id"`
	List     *string   `json:"list"   db:"list"`
	TagsJSON []byte    `json:"-"      db:"tags"`
	Tags     []string  `json:"tags"   db:"-"`
	Change   int       `json:"change" db:"change"`
}

File: cordova/src/js/model/series.ts
import { Model, ModelArray, prop, table } from 'js/model/model'
import Book from './book'

export type List = 'PLANNING' | 'READING' | 'COMPLETED' | 'PAUSED' | 'DROPPED'

@table('series', 'series', 'SeriesInput!', 'name', 'String!')
export default class Series extends Model {

    @prop('[Book]', { jsType: Book })
    public books: ModelArray<Book>

    @prop('List')
    public list: List

    @prop('[String]')
    public tags: string[]

    @prop('String')
    public name: string

    @prop('Int')
    public read: number

    @prop('Int')
    public total: number

    public get id() {
        return this.name
    }

    public get link() {
        return `/series/${this.name}`
    }

    public get sortIndex() {
        return `series-${this.name}`
    }
}
